runelite-client/src/main/java/net/runelite/client/plugins/config/ConfigPanel.java

```diff
@@ -72,75 +71,106 @@
 	private static final int SPINNER_FIELD_WIDTH = 6;
 
 	private final ConfigManager configManager;
-	private JTextField searchBar;
+	private JTextField searchBar = new JTextField();
+	private Map<String, JPanel> children = new TreeMap<>();
+	private int scrollBarPosition = 0;
 
 	public ConfigPanel(ConfigManager configManager)
 	{
 		super();
 		this.configManager = configManager;
-		populateConfig();
-	}
-
-	@Override
-	public void invalidate()
-	{
-		super.invalidate();
 
-		if (searchBar != null)
+		searchBar.getDocument().addDocumentListener(new DocumentListener()
 		{
-			searchBar.requestFocusInWindow();
-		}
+			@Override
+			public void insertUpdate(DocumentEvent e)
+			{
+				onSearchBarChanged();
+			}
+
+			@Override
+			public void removeUpdate(DocumentEvent e)
+			{
+				onSearchBarChanged();
+			}
+
+			@Override
+			public void changedUpdate(DocumentEvent e)
+			{
+				onSearchBarChanged();
+			}
+		});
+
+		rebuildPluginList();
+		openConfigList();
 	}
 
-	private void populateConfig()
+	void rebuildPluginList()
 	{
-		removeAll();
-		add(new JLabel("Plugin Configuration", SwingConstants.CENTER));
-		searchBar = new JTextField();
-		add(searchBar);
-		final Map<String, JPanel> children = new TreeMap<>();
-
+		Map<String, JPanel> oldChildren = children;
```

Comment:
Instead of replacing references, create copy of map, then clear original children